gcovr/coverage.py

```diff
@@ -90,6 +90,15 @@ def coverage_stat(key: str) -> CoverageStat:
             return cov.decision_coverage()
         return cov.line_coverage()
 
+    def key_num_lines(key: str) -> int:
+        stat = coverage_stat(key)
+        # When sorting in ascending order, a heuristic not uncommonly
+        # employed when looking to improve coverage, is to browse
+        # the list from worst to best, so it makes sense to place
+        # files with many lines higher up in groups that otherwise share
+        # the same (primary) sorting key.
+        return -stat.total
+
```

Comment:
I do not agree with this. If the numbers are the same the alphabetical order is better than sorting them by the total number of branches or lines.

Reply:
I think there are two camps here, or at least it depends on what your intentions with the report are.  
I'd find the sorting useful if I were looking for where adding tests would give most bang for the buck (especially those files with zero coverage) or understand which files contribute most to the overall coverage, but I can also see how sorting files based on their path/name is useful for other use cases.  
That said, I'll leave the final decision to you.

Reply:
I would remove this level of sorting again. Imagine someone uses `--sort uncovered-number` and the second key is the total number of lines if the uncovered number of lines is the same, the file with more lines will come first but has a better coverage in %.